## Problem Set 1

### Question 1 C#
#### What data type is each of the following ?

1) 5 


2) 5.0

3) 5 > 1

4) '5'

5) 5 * 2

6) '5' * 2

7) '5' + '2'

8) 5 / 2

9) 5 % 2

10) { 5, 2, 1 }

11) 5 == 3

12) Pi { the number }

### Solution 1 C#
1)   5                  -> integer

2)   5.0                -> float

3)   5 > 1              -> boolean   

4)   '5'                -> char

5)   5 * 2              -> integer

6)   '5' * 2            -> integer 

7)   '5' + '2'          -> integer

8)   5 / 2              -> float

9)   5 % 2              -> integer

10)   { 5, 2, 1 }        -> object   

11)  5 == 3            -> boolean

12)  Pi { the number } -> double

========================================================================================================================

### Question 2 C#
#### Write and evaluate C# expressions that answer this questions:


#### a. How many letters are there in "Supercalifragilisticexpialidocious"

#### Solution 2-a C#

In [1]:
string word = "Supercalifragilisticexpialidocious";
int count = 0;
foreach (char c in word)
{
   count++;
}
return count;

The below script needs to be able to find the current output cell; this is an easy method to get it.

34

#### Solution 2-a Output:

34

=========================================================================================================================

#### b. Does 'Supercalifragilisticexpialidocious' contain 'ice' as a substring

#### Solution 2-b C#

In [9]:
string givenString = "Supercalifragilisticexpialidocious";
string checkString = "ice";
 
int occurrence = givenString.Split(checkString).Length - 1;

if (occurrence > 0) {
    Console.WriteLine($"Yes, {givenString} contain {checkString} as a substring");
} else {
    Console.WriteLine($"No, {givenString} does not contain {checkString} as a substring");
}

Yes, Supercalifragilisticexpialidocious contain ice as a substring


#### Solution 2-b Output:

Yes, Supercalifragilisticexpialidocious contain ice as a substring

=========================================================================================================================

#### c. Which of the following words is the longest: Supercalifragilisticexpialidocious, Honorificabilitudinitatibus, Bababadalgharaghtakamminarronnkonn

#### Solution 2-c C#

In [2]:
string line = "Supercalifragilisticexpialidocious, Honorificabilitudinitatibus, Bababadalgharaghtakamminarronnkonn";
string[] words = line.Split(new[] { ", " }, StringSplitOptions.None);
string word = "";
int ctr = 0;

Array.Sort(words);
List<string> myList = new List<string>();
foreach (String s in words)
{
    if (s.Length >= ctr)
    {
        word = s;
        ctr = s.Length;
        myList.Add(s);
    }
}

foreach (var item in myList)
{
    Console.WriteLine(item);
}

Bababadalgharaghtakamminarronnkonn
Supercalifragilisticexpialidocious


#### Solution 2-c Output

Bababadalgharaghtakamminarronnkonn

Supercalifragilisticexpialidocious

=========================================================================================================================

#### d. Which composer comes first in the dictionary: 'Beriloz', 'Borodin', 'Brian', 'Bartok', 'Bellini', 'Buxtehude', 'Bernstein'. Which one comes last?

#### Solution 2-d C#

In [4]:
String[] dictionaryNames = {"Belioz", "Borodin", "Brian", "Bartok", "Bellini", "Buxtehude", "Bernstein"};

Array.Sort(dictionaryNames);

Console.WriteLine("First Composer in the dictionary: {0}", dictionaryNames[0]);
Console.WriteLine("Last Composer in the dictionary: {0}", dictionaryNames[dictionaryNames.Length - 1]);

First Composer in the dictionary: Bartok
Last Composer in the dictionary: Buxtehude


#### Solution 2-d Output

First Composer in the dictionary: Bartok

Last Composer in the dictionary: Buxtehude

===============================================================================================================================

### Question 3 C#
#### Implement function triangleArea(a, b, c) that takes as input the length of the 3 sides of a triangle and returns the area of the triangle using Heron's formula.

#### Solution 3 C#

In [10]:
static void triangleArea(int a, int b, int c)
{
     double s = (a + b + c) / 2;
     double Area = Math.Sqrt(s * (s - a) * (s - b) * (s - c));
     Console.WriteLine(Area);
}

triangleArea(2, 2, 2);

1.7320508075688772


#### Solution 3 Output

1.7320508075688772

===============================================================================================================================

### Question 4 C#

#### Write a programm in C# Sharp to separate odd and even integers in separate arrays.
#### Go to the editor

#### Test Data:
#### Input the number of elements to be stored in the array: 5
#### Input 5 elements in the array:
#### element - 0 : 25
#### element - 1 : 47
#### element - 2 : 42
#### element - 3 : 56
#### element - 4 : 32

#### Solution 4 C#

In [6]:
int[] numberArray = new int[] {25, 47, 42, 56, 32};
int[] evenArray = new int[5];
int[] oddArray = new int[5];

int i, j = 0, k = 0;

for (i = 0; i < 5; i++) {
    if (numberArray[i] % 2 == 0) {
       evenArray[j] = numberArray[i];
       j++;
    } else {
       oddArray[k] = numberArray[i];
       k++;
    }
}
Console.WriteLine("The Even elements are:");
for (i = 0; i < j; i++) {
  Console.WriteLine(evenArray[i]);
}
Console.WriteLine("The Odd elements are:");
for (i = 0; i < k; i++) {
    Console.WriteLine(oddArray[i]);
}

The Even elements are:
42
56
32
The Odd elements are:
25
47


#### Solution 4 Output:

The Even elements are:

42

56

32

The Odd elements are:

25

47

============================================================================================================================

### Question 5 C#
#### a. Write a function inside(x,y,x1,x2,y1,y2) that returns true or false depending on whether the point (x,y) lies in the rectangle with lower left corner (x1,y1) and upper right corner (x2,y2).

#### Solution 5-a C#

In [5]:
static bool inside(int x, int y, int x1, int y1, int x2, int y2) 
{
    if (x >= x1 && x <= x2 && y >= y1 && y <= y2) 
    {
        Console.WriteLine("True");
        return true;
    } 
    else 
    {
        Console.WriteLine("False");
        return false;
    }
}

inside(1, 1, 0, 0, 2, 3);
inside(-1, -1, 0, 0, 2, 3);

True
False


#### Solution 5-a Output

True

False

========================================================================================================================

#### b. Use function inside() from part a. to write an expression that tests whether the point (1,1) lies in both of the following rectangles: one with lower left corner (0.3, 0.5) and upper right corner (1.1, 0.7) and the other with lower left corner (0.5, 0.2) and upper right corner (1.1, 2)

#### Solution 5-b C#

In [6]:
static bool inside(double x, double y, double x1, double y1, double x2, double y2) 
{
    if (x >= x1 && x <= x2 && y >= y1 && y <= y2) 
    {
        Console.WriteLine("True");
        return true;
    } 
    else 
    {
        Console.WriteLine("False");
        return false;
    }
}

inside(1, 1, 0.3, 0.5, 1.1, 0.7);
inside(1, 1, 0.5, 0.2, 1.1, 2);

False
True
